src/config.js

```diff
@@ -1239,6 +1408,87 @@ class Config {
       return new ScopeDescriptor({scopes})
     }
   }
+
+  collectFilePromises(configPaths) {
+    return configPaths.map((curPath) => {
+      const jsonPath = path.join(curPath, '.atom', 'config.json')
+      const csonPath = path.join(curPath, '.atom', 'config.cson')
+
+      return new Promise((resolve, reject) => {
+        fs.access(jsonPath, fs.constants.R_OK, (err) => {
```

Comment:
is there already an abstraction in atom over promisifying `fs` and other errback modules?

Reply:
I'm not sure - although I know this exists in Nuclide.